src/script/descriptor.cpp

```diff
@@ -312,6 +323,18 @@ class BIP32PubkeyProvider final : public PubkeyProvider
         }
         return true;
     }
+    bool GetPrivKey(int pos, const SigningProvider& arg, CKey& key) const override
+    {
+        CExtKey extkey;
```

Comment:
In commit "Add private key derivation functions to descriptors" (ca039db0776b1626ce4d220a0da5ff985d93985d)

This is duplicating code already in GetPubKey above. Since this has to exist separately, it'd be good to change GetPubKey to call this function. It'd simplify GetPubKey and remove a chunk of duplicate code.